urllib3/request.py

```diff
@@ -61,91 +54,79 @@ def request(self, method, url, fields=None, headers=None, **urlopen_kw):
         :meth:`request_encode_url`, :meth:`request_encode_body`,
         or even the lowest level :meth:`urlopen`.
         """
+        pops = [
+            'encode_multipart',
+            'multipart_boundary',
+            'form_fields',
+            'url_params',
+            'fields'
+        ]
         method = method.upper()
-
-        if method in self._encode_url_methods:
-            return self.request_encode_url(method, url, fields=fields,
-                                           headers=headers,
-                                           **urlopen_kw)
-        else:
-            return self.request_encode_body(method, url, fields=fields,
-                                            headers=headers,
-                                            **urlopen_kw)
-
-    def request_encode_url(self, method, url, fields=None, headers=None,
```

Comment:
Do you want to re-insert the existing code, or would shims that have the same behavior, using the new methods, work?